src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,130 @@
+import { IconId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import { Fragment, useEffect, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { ActionBar } from 'src/components/ActionBar';
+import { Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { SimpleTable } from 'src/components/table';
+import { StringInput } from 'src/data-catalog/create-dataset/CreateDatasetInputs';
+import { GetConceptsResponse } from 'src/dataset-builder/DatasetBuilderUtils';
+import { SnapshotBuilderConcept as Concept, SnapshotBuilderDomainOption } from 'src/libs/ajax/DataRepo';
+import { DataRepo } from 'src/libs/ajax/DataRepo';
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import colors from 'src/libs/colors';
+
+import { PAGE_PADDING_HEIGHT, PAGE_PADDING_WIDTH } from './constants';
+
+type ConceptSearchProps = {
+  readonly initialSearch: string;
+  readonly domainOption: SnapshotBuilderDomainOption;
+  readonly onCancel: () => void;
+  readonly onCommit: (selected: Concept[]) => void;
+  readonly onOpenHierarchy: (
+    domainOption: SnapshotBuilderDomainOption,
+    selected: Concept[],
+    searchText: string
+  ) => void;
+  readonly actionText: string;
+  readonly datasetId: string;
+  readonly initialCart: Concept[];
+};
+
+export const ConceptSearch = (props: ConceptSearchProps) => {
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  return h(Fragment, [
+    div({ style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` } }, [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: onCancel,
+            'aria-label': 'cancel',
+          },
+          [icon('left-circle-filled', { size: 32 })]
+        ),
+        div({ style: { marginLeft: 15 } }, [domainOption.category]),
+      ]),
+      h(StringInput, {
+        onChange: (value: string) => {
+          setSearch(value);
+        },
+        value: search,
+        placeholder: 'Search',
+      }),
+      concepts.status === 'Ready'
+        ? h(SimpleTable, {
+            'aria-label': 'concept search results',
+            underRowKey: 'underRow',
```

Comment:
Do we have an underrow?

Reply:
No, but the code wouldn't compile without this, and adding this doesn't seem to break anything (since the row lookup in `SimpleTable` returns undefined). I did take a quick look at trying to fix this in `SimpleTable` (which is not yet typescript) and wasn't sure how to do it.